services/QuillLMS/app/models/unit_template.rb

```diff
@@ -107,12 +107,12 @@ def self.user_scope(user_flag)
     USER_SCOPES.fetch(user_flag, UnitTemplate.production)
   end
 
-  def get_cached_serialized_unit_template(flag=nil)
+  def get_cached_serialized_unit_template(flag=nil, current_user=nil)
     cache_expiration_time = 600
     cached = $redis.get("unit_template_id:#{id}_serialized")
     serialized_unit_template = cached.nil? || cached&.blank? ? nil : JSON.parse(cached)
     unless serialized_unit_template
-      serializable_unit_template = UnitTemplatePseudoSerializer.new(self, flag)
+      serializable_unit_template = UnitTemplatePseudoSerializer.new(self, flag, current_user)
       serialized_unit_template = serializable_unit_template.data
       $redis.set("unit_template_id:#{id}_serialized", serialized_unit_template.to_json, {ex: cache_expiration_time})
     end
```

Comment:
I'm pretty sure this setup is going to cache, for all users, the unit template data for any given user (because the cache key itself doesn't care about the user). My inclination is to say that we should probably split that data out into a separate query and maybe a separate controller action, so we can leave the caching logic alone, but I'd defer to a backend dev on that front.

Reply:
ahh ok, yeah I didn't think about the caching aspect. I agree, I'll pull the logic out into a separate query and controller action